ENIGMAsystem/SHELL/Widget_Systems/xlib/kdialog.cpp

```diff
@@ -192,7 +184,7 @@ static inline void show_debug_message_helper(string errortext, MESSAGE_TYPE type
   if (strtod(str_result.c_str(), NULL) == 1) exit(0);
 }
 
-namespace enigma_user {
+namespace kdialog {
```

Comment:
Now, these files are where the magic happens. This stops being a namespace and becomes a class:
```cpp
class KDialogWidgets : public CommandLineWidgetEngine {
 public:
  void show_debug_message(string errortext, MESSAGE_TYPE type) override {
    ...
  }
  ...
};
```
Notice that the word `override` is added at the end of each method. This prevents typos between the engines, requiring they all take the same arguments and become linked correctly.